### ISSUE TITLE
IE9 can only (pre)load max. 41 audio files; maybe also iPad affected

### ISSUE THREAD

**Issue by netmills** [CreateJS/SoundJS#40]
##### Undocumented IE behavior

IE9 (and IE10?) has an internal limit of 41 simultaneous audio and video elements/objects. The 42nd won't be loaded. 

It seems to me (without having tested it enough) that this is a "global" limit, affecting all open tabs, and not only a single html document. So it might be that one can load less than 41 audio/videos.
##### SoundJS/PreloadJS bug

PreloadJS with the SoundJS plugin creates an audio object for each loaded file. So if you load to many audio files, some will will not load - and the error event will be raised containing an valid HTMLAudioElement, which won't play. 

The sound will play at random on your site (depending which loaded first). Also videos won't play after preloading sounds.

I reckon that even without preloading SoundJS will not be able to play more than 41 sounds. 
I am not sure, but it seems that HTMLAudioPlugin keeps the played audio elements in memory.
##### Possible solution
- Don't create HTMLAudioElements when preloading (at least in IE). This would affect existing code of developers using the library.
- Reuse the max. 41 "slots" when playing sound on IE.
##### Further information

It took me almost a day to find this piece of information - more or less by accident :(
This is the only information i could find:
- See the [comment](http://www.gamasutra.com/blogs/ChrisKhoo/20120618/172573/The_HTML5_audio_troubleshooting_guide.php) of Kevin Gadd
- Or Barry Dormans [answer on stackoverflow](http://stackoverflow.com/questions/6033395/do-html5-audio-tags-eat-up-resources)
- Sam might have the [same issue](http://community.createjs.com/discussions/preloadjs/148-ie9-failing-to-load-audio-but-still-triggers-complete)

If you find an official documentation of this limit, I would be glad if you let me know.
##### iPad bug?

We also found a very weird behavior on the iPad (with WebAudioPlugin). At some point our sounds sounded completely distorted and also played to slow. (The sound files are fine, it also occurs with sound files from you demos.) At the beginning of the project everything was ok (we used less sounds them).

I also assume that we are preloading to many sound files, but I don't have the time to isolate the error. (If one sound is distorted, subsequently sound may also be - even in other tabs. You must then quit (not only close) Mobile Safari and delete the cache to have a "clean" state for testing.)

Sorry, but currently I have no time to create a test page. If you need one it will take some time.

**Comment by Tolmark12** [comment]
Does anyone know if this affects IE 11?

**Comment by NuclearCookie** [comment]
Hello. 
I found no concrete answers if this was still an issue in IE 11. 
I created a JSFiddle to test it out. 
Turns out it's still there, but 41 does not seem to be a magic number. I get inconsistent numbers of failures. ( but always failures )
http://jsfiddle.net/NuclearCookie/6c1dps3b/

**Comment by themoonrat** [comment]
This still affects IE11, yes. And there isn't a magic number for it at all, it seems rather arbitrary. In testing (getting the whole office involved!) I found that I was always _**definitely**_ safe at 20, and, apart from very rare cases, almost certainly safe at 25. The higher the amount after that, the greater the risk of some not loading.

So for my games, I tended to list first 25 sounds in the manifest as the most important ones, and any after that being spot effects that, if they didn't load and play, wouldn't be the end of the world.

For more recent games, I've been able to take advantage of the sound sprite feature within SoundJS, to get more sound samples within the same amount of audio files.
